### PR TITLE
[1.21] Move Mod button below Realms and adjust branding text

### LINKED ISSUE #249: Improve the position of the "Mods" menu in Minecraft's title screen
Currently, it looks like this:

image

image

The "Minecraft Realms" menu is smaller, so the icons get stuck in the letters.

I think it would be possible to find a position in the title screen better adapted to the mod menu.

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
The goal of this PR is to fix #249 issue and supersede https://github.com/neoforged/NeoForge/pull/456 PR. The patch is smaller with this PR.

What is done is the Mod button is moved below the Realms button so both can be full width. Then the mod count in lower left corner was removed and instead, the mod count is now part of the Mod button text. Let me know if I done the lang file changes incorrectly.

Here's what it looked like on various screen sizes. Note, the NeoForge text in bottom left is super long only in PR branches. In prod, it would be `NeoForge-20.6.76-beta` which is much shorter and won't overlap the Mojang branding text in bottom right.

image
image
image
image
image

### CODE REVIEW

**Comment by TelepathicGrunt** [comment]
Slight change now. Moved all menu buttons up 12 pixels to keep it from feeling squished at bottom. Will not overlap with bottom text ever now. Can do 6 pixels instead if too high now

image
image

**Comment by TelepathicGrunt** [comment]
image
image-1
image

**Comment by TelepathicGrunt** [comment]
Note to self, add mod count to pause menu's mod button

**Comment by TelepathicGrunt** [comment]
Pause menu mod button now fixed to show mod count
image

**Comment by HenryLoenwind** [comment]
I like the Mod button being half-sized on the main menu. It makes it stand out, and it doesn't get lost in a long list of identical buttons.

For reference:
image

There is a grouping there that makes each button stand on their own. No matter from what angle you look at, each one has its distinct personality that makes it easy and quick to find. "The big top one", "The centre one", "The left and right ones of the top group", "The two footer ones", "The tiny icons to the sides you don't see unless you're looking for them".

In comparison, the new suggestion gives you a visual unity that makes the individual buttons disappear as they are mushed together into a list where you have to read the labels to identify them. The pause menu is even worse with its button overload (but that's not this PR's fault). Nothing really stands out there aside from the top and bottom buttons.

Showing the number of mods on the button has no precedence in Minecraft. As such it is unclear to the player what that number want to tell them. There's a real danger that they will unconsciously downgrade the button to an information label instead of identifying it as the place the mod configuration lives. If anything, I'd propose to change it to "Mods..." or even "Mod Options..." to make clear that there's a sub-menu behind the button. It doesn't start a process or execute a single action, so it should have the "..." marker.

For the mod count, I'd suggest "Minecraft 1.21 with 4 mods" instead. There's plenty of space on that line, and semantically, it also fits because those two things together determine your game experience.

---

As a reminder about UI design: Shapes, positions and colours are way more important than text. Most people remember UI elements based on that and not the written text. The text allows them to initially get acquainted with an element, but it is not the primary means to recognise it later. To close a window, you click on whatever is (on Windows) in the top-right corner, no matter if the icon is an X or not. The "snooze" button on your alarm clock is the space-bar-shaped one---do you remember where the label for that one is on your alarm clock? Or if it really says, "snooze"? Maybe it says, "Snooze!"? Or "打盹"? Does the volume-up key on your TV remote say "Vol-+" or "+" or "🔊+" or "🔊^"?

**Comment by TelepathicGrunt** [comment]
Mod button cannot be half size because it breaks realms button and makes it look like garbage due to overlapping icons and text. Which is worse design imo

image

**Comment by HenryLoenwind** [comment]
Then either put those icons, whatever they are, to the right of the button where there's plenty of space, or shorten the caption to "Realms" and left-align it when there are icons, or move the Mods button into the Options menu as "Mod Options" where it semantically belongs, or move the Realms button into the Multiplayer menu where it semantically belongs. There are plenty of ways of not sacrificing usability of the altar of beauty.

PS: didn't Realms have a check that locked out modded clients anyway, or did that change? I never looked into that since they added it, tbh.

**Comment by TelepathicGrunt** [comment]
Realms works with modded clients. And no, we aren't gonna have floating icons to right of button. That'll look awful. And no, even just "Realms" alone overlaps the icons uness you left align the realms text which then looks awful and inconsistent. If you are speaking about putting design foremost than anything else, then these suggestions violates "good" design

**Comment by HenryLoenwind** [comment]
image
 Looks like there's plenty of space, either left-aligned or centred in the remaining space...

**Comment by TelepathicGrunt** [comment]
That image you showed left aligned looks so bad. It’s inconsistent with the design of all other buttons. And I can see clearly if centered, the right side of realms will overlap the blue icon. 

This is not the way

**Comment by TelepathicGrunt** [comment]
To be clear, I’m going to take some inspiration for Mod Menu on fabric to try some of their decisions but I really do not like how they replaced the feedback/bugs buttons entirely on the pause menu. 

The mod count on the mod button should be fine. If the count is misleading, it would’ve been misleading anyway in bottom left corner. I’ll think about removing it but having it upfront might help reduce number of people asking why a mod isn’t working when it isn’t even on. Since they would see that their mod count didn’t go up upon adding a mod and know now to check their mods folder for if they grabbed the right jar and put it in right spot. I had a few cases where users didn’t realize the mod wasn’t on at all

**Comment by IMS212** [comment]
I believe replacing the feedback button is not only justified, but a good idea; modded users should not be reporting bugs to Mojang on a modded client ever.

**Comment by Gaming32** [comment]
Since 1.21, the feedback button is configurable by the server.

**Comment by Prospector** [comment]
> To be clear, I’m going to take some inspiration for Mod Menu on fabric to try some of their decisions but I really do not like how they replaced the feedback/bugs buttons entirely on the pause menu.

In Mod Menu, these buttons are available inside the mods screen so they are redundant. Unsure if NeoForge does the same. In fact, I would think the report bugs button especially should *not* be a top-level button if you're running mods, because the bugs you encounter are less likely to be ones you should report to Mojang.

Also, including the mod count in the button is something I did a long time ago, and I had it formatted how you did in an early alpha version like 7 years ago. In the end, it was really inconsistent with how other buttons are labeled and people didn't like it a whole lot. In the end, I settled for making the default just say "Mods" with the count in the lower corner.

**Comment by Gaming32** [comment]
> because the bugs you encounter are less likely to be ones you should report to Mojang

Again, in the case of a server, this button does not necessarily go to Mojang, but rather may go to the server's own bug report page, as defined by `bug-report-link` in `server.properties`. In addition, this button may be replaced by a "Server Links" button with arbitrary links if the `minecraft:server_links` packet is sent by the server.

**Comment by Prospector** [comment]
No, the report bugs button only ever goes to Mojang's URL (hardcoded to Urls.SNAPSHOT_BUGS in yarn), but it gets replaced by "Server Links" if the server provides that. Mod Menu does not replace or hide "Server Links", only the "Feedback", "Report Bugs", and "Give Feedback..." buttons which all only go to Mojang stuff, which is already inside Mod Menu under "Minecraft"

**Comment by TelepathicGrunt** [comment]
For reference, this is how serverlinks works:
image

**Comment by TelepathicGrunt** [comment]
Hmm
image
image
image

**Comment by TelepathicGrunt** [comment]
What we are setting on
image
image

**Review by sciwhiz12 — DISMISSED** [review]
Will merge in a few hours, to await any final feedback.

**Review by Technici4n — DISMISSED** [review]
Didn't check the code but I like the design.

File: src/main/java/net/neoforged/neoforge/internal/BrandingControl.java [hunk 1]
```diff
@@ -27,10 +25,8 @@ public class BrandingControl {
     private static void computeBranding() {
         if (brandings == null) {
             ImmutableList.Builder<String> brd = ImmutableList.builder();
-            brd.add(ForgeSnapshotsMod.BRANDING_NAME + ' ' + NeoForgeVersion.getVersion());
             brd.add("Minecraft " + DetectedVersion.BUILT_IN.getName());
-            int tModCount = ModList.get().size();
-            brd.add(FMLTranslations.parseMessage("fml.menu.loadingmods", tModCount));
+            brd.add(ForgeSnapshotsMod.BRANDING_NAME + ' ' + NeoForgeVersion.getVersion() + " (" + net.neoforged.fml.ModList.get().size() + " mods)");
```
Comment (HenryLoenwind) [inline src/main/java/net/neoforged/neoforge/internal/BrandingControl.java]: This needs to be translatable.

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**